js/mirador_viewer.js

```diff
@@ -12,17 +12,22 @@
      */
     Drupal.behaviors.Mirador = {
         attach: function (context, settings) {
+            Drupal.IslandoraMirador = Drupal.IslandoraMirador || {}
+            Drupal.IslandoraMirador.instances = Drupal.IslandoraMirador.instances || {}
             Object.entries(settings.mirador.viewers).forEach(entry => {
               const [base, values] = entry;
               once('mirador-viewer', base, context).forEach(() =>
-                Mirador.viewer(values, window.miradorPlugins || {})
+                // save the mirador instance so other modules can interact
+                // with the store/actions at e.g. Drupal.IslandoraMirador.instances["#mirador-xyz"].store
+                Drupal.IslandoraMirador.instances[base] = Mirador.viewer(values, window.miradorPlugins || {})
               );
             });
         },
         detach: function (context, settings) {
             Object.entries(settings.mirador.viewers).forEach(entry => {
               const [base, ] = entry;
               once.remove('mirador-viewer', base, context);
+              delete Drupal.IslandoraMirador.instances[base];
```

Comment:
This would always delete the item, independent if it was actually contained/affected in `context`.

Instead, probably something like:

```suggestion
              const removed = once.remove('mirador-viewer', base, context);
              if (removed.length > 0) {
                delete Drupal.IslandoraMirador.instances[base];
              }
```

Strictly, given we expect `base` to be an HTML id (which should be unique), there should only ever be `0` or `1` item in `removed`, but I don't anticipate any possible issues with using an inequality?